Assets/Mirror/Runtime/Transport/Discovery/Utility/ByteStreamer.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+
+namespace Assets.Scripts.Utility.Serialisation
+{
+    // I have not included all my streaming utilities as I have encyrption, CLZF2 compression and a custom binary formatter that handles Unity vectors.
+    // I wanted to keep the sample small!
+    static class ByteStreamer
+    {
+        public static byte[] StreamToBytes(Object target)
+        {
+            using (MemoryStream memoryStream = new MemoryStream())
+            {
+                BinaryFormatter binaryFormatter = new BinaryFormatter();
+                binaryFormatter.Serialize(memoryStream, target);
+                return memoryStream.ToArray();
+            }
+        }
+
+        public static object StreamFromBytes(byte[] data)
+        {
+            using (MemoryStream memoryStream = new MemoryStream(data))
+            {
+                BinaryFormatter binaryFormatter = new BinaryFormatter();
+                return binaryFormatter.Deserialize(memoryStream);
```

Comment:
BinaryFormatter.Deserialize is a total security nightmare,  and in my opinion the biggest mistake ever made in .Net and java.

Note that you don't specify what class you are expecting.  The stream is expected to contain the name of the class that should be instantiated.   A malicious client can provide the name of _any_ class here,  and C# would call it's constructor here.   Basically this exposes **every** constructor reachable by the CLR to all the applications in the network.

A better way is to use JsonUtility.FromJson<T>  from Unity,  or manually deserialize the data using NetworkReader